js/modulestest/runtime.go

```diff
@@ -0,0 +1,51 @@
+// Package modulestest contains helpers to test js modules
+package modulestest
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/eventloop"
+	"go.k6.io/k6/lib"
+)
+
+// Runtime is a helper struct that contains what is needed to run a (simple) module test
+type Runtime struct {
+	VU            *VU
+	EventLoop     *eventloop.EventLoop
+	CancelContext func()
+}
+
+// NewRuntime will create a new test runtime and will cancel the context on test/benchmark end
+func NewRuntime(t testing.TB) *Runtime {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+	vu := &VU{
+		CtxField:     ctx,
+		RuntimeField: goja.New(),
+	}
+	vu.RuntimeField.SetFieldNameMapper(common.FieldNameMapper{})
+	eventloop := eventloop.New(vu)
+	vu.RegisterCallbackField = eventloop.RegisterCallback
+	result := &Runtime{
+		VU:            vu,
+		EventLoop:     eventloop,
+		CancelContext: cancel,
+	}
+	// let's cancel again in case it has changed
+	t.Cleanup(func() { result.CancelContext() })
+	return result
+}
+
+// InitContext will set the InitContext of the Runtime
+func (r *Runtime) InitContext(i *common.InitEnvironment) {
+	r.VU.InitEnvField = i
+}
+
+// MoveToVUContext will set the state and nil the InitEnv just as a real VU
+func (r *Runtime) MoveToVUContext(state *lib.State) {
```

Comment:
Is this a simplified version of the activate operation? If yes, then I would prefer some sort of parallelism with the real VU in terms of naming.

Reply:
That doesn't sound like a bad idea ... but I am not certain that someone who isn't aware of the particulars of the k6 internals will be more okay with `Activate`. 

Arguably that name exists more because we are trying to hide the internals of the js.Runner outside of it. But here we are basically testing it/around it/in it. So it seems more useful IMO to name this something that the user (writing the test) will know they need. And most users are aware that they need the `*lib.State` in order to execute most of their code.

Maybe if decide to have overarching naming of all the phases we can use those - but that is way bigger task then this PR :) 


WDYT @na-- @oleiade

Reply:
I have no strong opinions on that matter. TL;DR: I'm rather for keeping it named `MoveToVUContext`. 

When trying out the integration in xk6-redis, I found `MoveToVUContext` quite intuitive, as making the runtime enter the VU context, or the init context, are basically the two only operations I was interested in.

I think `Activate` would have been confusing for me, but I haven't used much that part of the k6 internals either, so I might have a colder cache than you folks on that front. I would have judged `Activate` taken out of the k6 context as to broad, "activate, what? why?", whereas "move (what: the runtime) into a VU context" feels self-sufficient (although the word context is so polysemic in Go that it can lead to confusion.

Reply:
>When trying out the integration in xk6-redis, I found MoveToVUContext quite intuitive

Out of curiosity, why VUContext is so intuitive for you? We don't mention in any doc the concept about `VUContext` and also the code doesn't export an API mentioning it. I guess is a concept that has [some historical reasons](https://github.com/grafana/k6/commit/a0b0eb12c994fec1fd83c565b2fcb443361166b1) because the unique place where we really mention is here: https://k6.io/docs/javascript-api/init-context

Reply:
I guess it's because from a purely programming perspective, it's named exactly after what I wanted to perform: "Prepare a runtime that's set up to reproduce the context of a VU executing a script". Probably auto-complete factors in: if I type `myRuntime.` and look at the auto-completion, `MoveToVUContext` lets me know somewhat clearly what's going to happen, whereas `Activate`, doesn't answer fundamental questions such as "what is implied by activate? What does it activate?", it's quite vague, and the expectation is unclear. But arguably, when programming (as in writing tests), I don't use the documentation, I only rely on the code's documentation, and the auto-completion mostly.

Reply:
Resolving this as it seems as there is consensus